breakingchanges.md

```diff
@@ -0,0 +1,153 @@
+- `set` and `frozenset` now share much of their implementation
+- `collections` module rewritten using new api
+- `itertools` module rewritten using new api - these are now type objects rather than instances of `generator`
+- `dict` and `set` throw errors if the objects change size during iteration as per Cpython.
+- `Sk.builtin.check*` moved to `src/check.js`
+- `enumerate`/`filter_`/`reversed`/`zip_`/`map_` combined into one file `src/iteratorobjects.js`
+- `tuple_iter_`/`str_iter_` etc combined into `src/simple_iterators.js`
+- `dictviews` like `dict_items` etc moved to `src/dictviews.js`
+- `number.js` removed
+- `numtype.js` removed
+- `seqtype.js` removed
+- `Sk.builtin.check*` moved to `src/check.js`
+- `mp$subscript` should not be called by a js object (see changes in `random.js`)
+
+
+**call signatures of builtins**
+- `new` is required for (almost) all builtin types
+  - 3 exceptions - `Sk.builtin.bool`, `Sk.builtin.none`, `Sk.builtin.NotImplemented`
+  - These 3 will always return their respective constant(s) and are thus not required to be used as constructors. 
+- Restricted parameters for directly accessing a constructor of an `Sk.builtin` type
+
+
+
+| type  | params   | notes |
+|---|---|---|
+| `Sk.builtin.int_`  | `{number| JSBI (bigint)| string| undefined}`  | |
+| `Sk.builtin.float_`  | `{number}`  | |
+| `Sk.builtin.complex`  | `{number, number}`  | |
+| `Sk.builtin.list`  | `{Array=}`  | |
+| `Sk.builtin.tuple`  | `{Array=}`  | |
+| `Sk.builtin.set`  | `{Array=}`  | |
+| `Sk.builtin.dict`  | `{Array=}`  | key/value pairs - only pyObjects |
+| `Sk.builtin.str`  | `{*}`  | can be used to convert a pyObject|
+| `Sk.builtin.bool`  | `{*}`  | can be used to convert a pyObject|
+
+
+**Major changes**
+- All type objects are now callable using their respective `tp$call` methods inherited from `Sk.builtin.type`
+- All native type objects will require a `tp$new` and `tp$init` method (maybe inherited by `Sk.builtin.object`)
+- All type objects are javascript instances of `Sk.builtin.type`
+- All single inherited objects follow javascript inheritance
+- All native type objects now have the following and replaces the use of `Sk.builtin.function` for all dunder function/methods.
+  - `wrapper_descriptors` aka `slot_wrappers`
+  - `method_descriptors` 
+  - `classmethod_descriptors`
+  - `getset_descriptors` aka `attributes`/`member_descriptors`
+- `Sk.builtin.sk_method` is an alternative to `Sk.builtin.function` and is used by the above `descriptor` types
+- mangled names are never passed to the user but instead are an attribute on `Sk.builtin.str` instances as `$mangled`
+- `mappingproxy` added
+- `$d` removed on all type objects.
+- `attributes` of a type object now only appear on the `prototype`. Previously these appeared on both the `type` object and the `prototype`
+
+
+
+**Additions**
+- `dict`, `tuple` are suspendable
+- `classmethod`, `property`, `staticmethod` have native skulpt implementations
+- `super` can now be unbound
```

Comment:
See this previously notImplementedError 

https://github.com/skulpt/skulpt/blob/master/src/typeobject.js#L56

And this:

https://stackoverflow.com/questions/30190185/how-can-i-use-super-with-one-argument-in-python/30190341#30190341

I commented some tests for this in test_super. Though the tests needed changing since they  were out of date for py3. I tested the tests in python3 to make sure they were accurate. I should probably comment the py2 tests back out and find the appropriate py3 tests and move these to unit3/

Reply:
to resolve this I added the stackoverflow link in the breaking changes. 
i updated 
`test_supers` in `unit/test_descr.py` with the up to date 2.7 tests
(this highlighted a small bug I had and fixed this)
I also included the cpython `test_super.py` file and uncommented all the tests that worked. 
commented tests were those that we don't yet support and one that I will raise an issue which is a closure issue.